app/src/main/java/org/mozilla/fenix/search/awesomebar/AwesomeBarView.kt

```diff
@@ -117,6 +120,14 @@ class AwesomeBarView(
                 engineForSpeculativeConnects
             )
 
+        historyMetadataProvider =
+            HistoryMetadataSuggestionProvider(
+                historyStorage =components.core.historyStorage,
+                loadUrlUseCase = loadUrlUseCase,
+                icons = components.core.icons,
+                engine = engineForSpeculativeConnects
```

Comment:
@Mugurell added a new parameter `maxNumberOfSuggestions` which we should set. The ticket (FNX-22340) wants us to display the top 3 results, so you could add `maxNumberOfSuggestions = 3`.

Reply:
This is waiting for AC bump.